### PR TITLE
Fix homogen builder reset with rvalue ref

### PR DESCRIPTION
**Pull request by rlnx**
- Update `homogen_table_builder::reset` as discussed with @michael-smirnov 
- Change pointer to `array` in `table_iface`

### CODE REVIEW

File: cpp/oneapi/dal/table/backend/homogen_table_impl.hpp [hunk 1]
```diff
@@ -71,8 +71,8 @@ class homogen_table_impl
         return meta_;
     }
 
-    const void* get_data() const override {
-        return data_.get_data();
+    array<byte_t> get_data() const override {
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/table/backend/homogen_table_impl.hpp:74]: You already have `get_data_array()` on 86 line. Probably, it is needed to be removed or propagated to the base interface like `get_data()`.

Comment (rlnx) [reply cpp/oneapi/dal/table/backend/homogen_table_impl.hpp:74]: Removed

**Review by KulikovNikita — APPROVED** [review]

**Pull request merged**